src/chart/graph/Thumbnail.ts

```diff
@@ -0,0 +1,204 @@
+
+        this._treeModel = seriesModel;
+
+        const childrenNodes = (this._parent.children()[0] as graphic.Group).children();
+        const symbolNodes = (childrenNodes[0] as graphic.Group).children();
+        const lineNodes = (childrenNodes[1] as graphic.Group).children();
+
+        const backgroundColor = model.getModel('backgroundColor');
+        const borderColor = model.getModel('borderColor');
+        const thumbnailWidth = model.get('width');
+        const thumbnailHeight = model.get('height');
+        const selectedDataBackground = model.getModel('selectedDataBackground');
+
+        this._layoutParams = {
+            pos: {
+                left: model.get('left'),
+                right: model.get('right'),
+                top: model.get('top'),
+                bottom: model.get('bottom')
+            },
+            box: {
+                width: api.getWidth(),
+                height: api.getHeight()
+            }
+        };
+
+          const layoutParams = this._layoutParams;
+
+          this._widthProportion = <number>thumbnailWidth / layoutParams.box.width;
+          this._heightProportion = <number>thumbnailHeight / layoutParams.box.height;
+
+          const thumbnailGroup = new graphic.Group();
+
+        for (const node of symbolNodes) {
+              const sub = (node as graphic.Group).children()[0];
+              const x = (node as SymbolClz).x;
+              const y = (node as SymbolClz).y;
+
+              const subThumbnail = new (sub as any).constructor({
+                  shape: {
+                        ...(sub as graphic.Path).shape,
+                        width: 5,
+                        height: 5,
+                        x: (x * 0.25 - 2.5),
+                        y: (y * 0.25 - 2.5)
+                      },
+                  style: (sub as graphic.Path).style,
+                  z: 3,
+                  z2: 151
+              });
+
+              thumbnailGroup.add(subThumbnail);
+        }
+
+      for (const node of lineNodes) {
+          const line = (node as graphic.Group).children()[0];
+          const lineThumbnail = new ECLinePath({
+              style: (line as ECLinePath).style,
+              shape: (line as ECLinePath).shape,
+              scaleX: 0.25,
+              scaleY: 0.25,
+              z: 3,
+              z2: 151
+          });
+          thumbnailGroup.add(lineThumbnail);
+      }
+
+      const thumbnailWrapper = new graphic.Rect({
+          style: {
+              stroke: borderColor.option,
+              fill: backgroundColor.option,
+              lineWidth: 2
+          },
+          shape: {
+              height: thumbnailHeight === 0 ? layoutParams.box.height / 4 : thumbnailHeight,
+              width: thumbnailWidth === 0 ? layoutParams.box.width / 4 : thumbnailWidth
+          },
+          z: 2,
+          z2: 130
+      });
+
+      this._wrapper = thumbnailWrapper;
+
+      const areaStyle = selectedDataBackground.get('areaStyle');
```

Comment:
It's not necessary to separate to `areaStyle` and `lineStyle`. Just use one `selectedBackgroundStyle` and use `getItemStyle` to map to zrender style